sync/sync.go

```diff
@@ -314,8 +316,10 @@ func (s *Syncer[H]) requestHeaders(
 			size = amount
 		}
 
+		now := time.Now()
```

Comment:
minor optimization thought: when metrics are disabled we do this for no reason. And this is calling time.Now, which, in fact, does a syscall to OS to get the time from it, which is expensive. Ideally, we would do s.metric.rangeRequestStart and Stop calls which would internally keep the time and call it only when enabled. We don't have to do this now, but a food for thought

Reply:
true. Can create an issue and do it in a subsequent PR.

Reply:
I have done it here 😅 . I had to resolve conflicts, so decided to fix it rn.